site/en/r2/tutorials/generative/dcgan.ipynb

```diff
@@ -1,143 +1,197 @@
 {
```

Comment:
<p>Use tfds?</p><p><br></p><p>(This adds a 30s delay,  🙁)</p><p><br></p><p>```</p><p>!pip install tfds_nightly</p><p>import tensorflow_datasets as tfds</p><p><br></p><p>train_dataset = tfds.load('mnist', split = tfds.Split.TRAIN)</p><p><br></p><p>def format_dataset(features):</p><p>return (features['image']-127.5)/127.5)</p><p><br></p><p>train_dataset = mnist.map(format_dataset)</p><p>```</p>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/docs/pull/332/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Reply:
cast to float32 too.

30 secs delay per epoch?

Reply:
I am seeing a 3 sec delay.